pkg/controller/storageprofile-controller.go

```diff
@@ -283,29 +309,87 @@ func (r *StorageProfileReconciler) computeMetrics() error {
 	if err := r.checkIncompleteProfiles(); err != nil {
 		return err
 	}
-	if err := r.checkDefaultVirtStorageClasses(); err != nil {
+	if err := r.checkDefaultStorageClasses(); err != nil {
 		return err
 	}
 
 	return nil
 }
 
-func (r *StorageProfileReconciler) checkDefaultVirtStorageClasses() error {
+func (r *StorageProfileReconciler) checkDefaultStorageClasses() error {
+	defaultStorageClassCount := 0
 	defaultVirtStorageClassCount := 0
 	storageClassList := &storagev1.StorageClassList{}
 	if err := r.client.List(context.TODO(), storageClassList); err != nil {
 		return err
 	}
 	for _, sc := range storageClassList.Items {
+		if sc.Annotations[cc.AnnDefaultStorageClass] == "true" {
+			defaultStorageClassCount++
+		}
 		if sc.Annotations[cc.AnnDefaultVirtStorageClass] == "true" {
 			defaultVirtStorageClassCount++
 		}
 	}
+	DefaultStorageClassesGauge.Set(float64(defaultStorageClassCount))
 	DefaultVirtStorageClassesGauge.Set(float64(defaultVirtStorageClassCount))
 
+	hasSmartCloneAndRWX := 0
+	if defaultStorageClassCount > 0 || defaultVirtStorageClassCount > 0 {
+		sc := cc.GetFallbackStorageClass(storageClassList, cdiv1.DataVolumeKubeVirt)
```

Comment:
I think you're already (unintentionally?) always looking up the virt default storage class
(always pass cdiv1.DataVolumeKubeVirt content type)

Reply:
Your fallback func prefers the virt default storage class, but returns the default SC if virt default is not set.

Reply:
I am saying that you already unintentionally implemented https://github.com/kubevirt/containerized-data-importer/pull/2998#discussion_r1403587024 since this `cc.GetFallbackStorageClass(storageClassList, cdiv1.DataVolumeKubeVirt)` will only bring you virt storage classes